File: java/src/test/java/mx/iteso/WordAdapterTest.java
package mx.iteso;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

import mx.iteso.adapter.IGoogleDoc;
import mx.iteso.adapter.WordAdapter;
import mx.iteso.utils.BackgroundImage;
import mx.iteso.utils.Format;

import java.awt.Font;

public class WordAdapterTest {

    private WordDocument wordDocument = new WordDocument(1);
    private IGoogleDoc newGoogleDoc = new WordAdapter(wordDocument);

    @Test
    public void testGetFont() {
        assertEquals(new Font("Calibri", Font.ITALIC, 12), this.newGoogleDoc.getFont());
    }

    @Test
    public void testGetStyle() {
        assertEquals(new Format().getFont(), this.newGoogleDoc.getStyle().getFont());
    }

    @Test
    public void testGetBackground() {
        assertEquals(new BackgroundImage().getImage().getUrl(), this.newGoogleDoc.getBackground().getImage().getUrl());
    }

    @Test
    public void testGetSharingPermissions() {
        this.newGoogleDoc.setSharingPermissions(1);
        assertEquals(1, this.newGoogleDoc.getSharingPermissions());
    }

}

File: python/src/unittest/python/WordAdapterTest.py

from WordDocument import *
from WordAdapter import *
from BackImage import *
from Format import *
from Font import *
import unittest


class WordAdapterTest(unittest.TestCase):

    def setUp(self):
        self.word_doc = WordDocument(1)
        self.google_doc = WordAdapter(self.word_doc)

    def testStyle(self):
        self.assertEquals(Format().getFont().getFontText(),
                          self.google_doc.getStyle()
                          .getFont().getFontText())

    def testBackground(self):
        self.assertEquals(BackgroundImage().getImage().get_url(),
                          self.google_doc
                          .getBackground().getImage().get_url())

    def testPermissions(self):
        self.assertEquals(1, self.google_doc.getPermissions())

    def testFont(self):
        self.assertEquals(Font('Arial', 11).getFontText(),
                          self.google_doc.getFont().getFontText())

